cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/FileLocalizer.java

```diff
@@ -86,13 +87,15 @@ public void run(String[] args) throws Exception {
           break;
         }
 
-        Location location = context.getLocationFactory().create(localFile.getURI());
+        Location localizedFile = new LocalLocationFactory().create(new File(localFile.getURI().getPath()).toURI());
+        fileFetcher.download(localFile.getURI(), localizedFile);
+
         Path targetPath = targetDir.resolve(localFile.getName());
 
         if (localFile.isArchive()) {
-          expand(location, targetPath);
+          expand(localizedFile, targetPath);
```

Comment:
In here, you actually can start expanding the file when downloading the bytes (via ZipInputStream). This means the download should probably be as `download(URI source, OutputStream os)`.